AI/Nullkiller/Engine/PriorityEvaluator.cpp

```diff
@@ -114,10 +116,32 @@ uint64_t getCreatureBankArmyReward(const CGObjectInstance * target, const CGHero
 	auto creatures = bankInfo->getPossibleCreaturesReward();
 	uint64_t result = 0;
 
-	for(auto c : creatures)
+	const auto& slots = hero->Slots();
+	ui64 weakestStackPower = 0;
+	if (slots.size() >= GameConstants::ARMY_SIZE)
+	{
+		//No free slot, we might discard our weakest stack
+		weakestStackPower = std::numeric_limits<ui64>().max();
+		for (const auto stack : slots)
+		{
+			vstd::amin(weakestStackPower, stack.second->getPower());
```

Comment:
stack.second could be a nullptr and game will crash

Reply:
Not really. Stack should always have defined creature type - if it doesn't and AI has them in their army, then something is already **very** wrong.